operators/pkg/instance-controller/container_logic.go

```diff
@@ -0,0 +1,175 @@
+package instance_controller
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+	crownlabsv1alpha2 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha2"
+	instance_creation "github.com/netgroup-polito/CrownLabs/operators/pkg/instance-creation"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func CreateGUIContainerPod(name string, namespace string, template *crownlabsv1alpha2.Environment, instanceName string, references []metav1.OwnerReference, examMode bool) (appsv1.Deployment, error) {
+
+	var examModeStr string
+	if examMode {
+		examModeStr = "true"
+	} else {
+		examModeStr = "false"
+	}
+
+	tmpXVolumeMounts := corev1.VolumeMount{
+		Name:      "x-tmp-dir",
+		MountPath: "/tmp/.X11-unix",
+	}
+
+	containers := []corev1.Container{
+		{
+			Name:  "novnc",
+			Image: "qcfe/novnc",
+			Ports: []corev1.ContainerPort{{
+				ContainerPort: 80,
+			}},
+			Env: []v1.EnvVar{{
+				Name:  "HIDE_NOVNC_BAR",
+				Value: examModeStr,
+			}, {
+				Name:  "HTTP_PORT",
+				Value: "6080",
+			}},
+		},
+		{
+			Name:  "novnc",
+			Image: "qcfe/novnc-c-websockify-nginx",
+		},
+		{
+			Name:  "tigervnc",
+			Image: "qcfe/tigervnc-server",
+			VolumeMounts: []v1.VolumeMount{
+				tmpXVolumeMounts,
+			},
+		},
+		{
+			Name:  name,
+			Image: template.Image,
+			VolumeMounts: []v1.VolumeMount{
+				tmpXVolumeMounts,
+			},
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					"cpu": resource.MustParse(instance_creation.ComputeCPURequests(
+						template.Resources.CPU,
+						template.Resources.ReservedCPUPercentage,
+					)),
+					"memory": template.Resources.Memory,
+				},
+				Limits: v1.ResourceList{
+					"cpu": resource.MustParse(instance_creation.ComputeCPULimits(
+						template.Resources.CPU,
+						0, // no hypervisor
+					)),
+					"memory": template.Resources.Memory,
+				},
+			},
+		},
+	}
+
+	return appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            name + "-guicont-depl",
+			Namespace:       namespace,
+			OwnerReferences: references,
+			Labels: map[string]string{
+				"name":          name,
+				"template-name": template.Name,
+				"instance-name": instanceName,
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32Ptr(1),
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: containers,
+				},
+			},
+		},
+	}, nil
+}
+
+func (r *LabInstanceReconciler) CreateContainerEnvironment(
+	labInstance *crownlabsv1alpha2.Instance,
```

Comment:
```suggestion
	instance *crownlabsv1alpha2.Instance,
```

All references to the old `lab*` terminology should be avoided